pandas/core/indexes/base.py

```diff
@@ -296,10 +292,14 @@ def __new__(
 
         # categorical
         elif is_categorical_dtype(data) or is_categorical_dtype(dtype):
+            from pandas.core.indexes.category import CategoricalIndex
```

Comment:
Might be beneficial to add a comment detailing why these imports are delayed.

We used to delay these imports but looks like they got moved to the top as part of a cleaning/refactor, and would want to prevent that from happening again:
https://github.com/pandas-dev/pandas/blob/250574eda5302b6941f8c766f520f67226aadad7/pandas/core/indexes/base.py#L194-L198